src/test/fio_ceph_objectstore.cc

```diff
@@ -0,0 +1,345 @@
+/*
+ *  Ceph ObjectStore engine
+ *
+ * IO engine using Ceph's ObjectStore class to test low-level performance of
+ * Ceph OSDs.
+ *
+ */
+
+#include "os/ObjectStore.h"
+#include "global/global_init.h"
+
+#include <fio.h>
+
+struct fio_ceph_os_iou {
+	struct io_u *io_u;
+	int io_complete;
+};
+
+struct ceph_os_data {
+	struct io_u **aio_events;
+	ObjectStore *fs;
+};
+
+struct ceph_os_options {
+	struct thread_data *td;
+	char *objectstore;
+	char *filestore_debug;
+	char *filestore_journal;
+};
+
+// initialize the options in a function because g++ reports:
+//   sorry, unimplemented: non-trivial designated initializers not supported
+static struct fio_option* init_options() {
+	static struct fio_option options[] = {{},{},{},{}};
+
+	options[0].name     = "objectstore";
+	options[0].lname    = "ceph objectstore type";
+	options[0].type     = FIO_OPT_STR_STORE;
+	options[0].help     = "Type of ObjectStore to create";
+	options[0].off1     = offsetof(struct ceph_os_options, objectstore);
+	options[0].def      = "filestore";
+	options[0].category = FIO_OPT_C_ENGINE;
+	options[0].group    = FIO_OPT_G_RBD;
+
+	options[1].name     = "filestore_debug";
+	options[1].lname    = "ceph filestore debug level";
+	options[1].type     = FIO_OPT_STR_STORE;
+	options[1].help     = "Debug level for ceph filestore log output";
+	options[1].off1     = offsetof(struct ceph_os_options, filestore_debug);
+	options[1].category = FIO_OPT_C_ENGINE;
+	options[1].group    = FIO_OPT_G_RBD;
+
+	options[2].name     = "filestore_journal";
+	options[2].lname    = "ceph filestore journal path";
+	options[2].type     = FIO_OPT_STR_STORE;
+	options[2].help     = "Path for a temporary journal file";
+	options[2].off1     = offsetof(struct ceph_os_options, filestore_journal);
+	options[2].def      = "";
+	options[2].category = FIO_OPT_C_ENGINE;
+	options[2].group    = FIO_OPT_G_RBD;
+
+	return options;
+};
+
+/////////////////////////////
+
+
+struct OnCommitted : public Context {
+  struct io_u *io_u;
+  OnCommitted(struct io_u* io_u) : io_u(io_u) {}
+  void finish(int r) {
+  }
+};
+
+struct OnApplied : public Context {
+  struct io_u *io_u;
+  ObjectStore::Transaction *t;
+  OnApplied(struct io_u* io_u, ObjectStore::Transaction *t) : io_u(io_u), t(t) {}
+  void finish(int r) {
+
+	struct fio_ceph_os_iou *fio_ceph_os_iou =
```

Comment:
The indent looks doesn't right.

Could you pls fix it follow 

// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
// vim: ts=8 sw=2 smarttab